docs/enhancements/extract-os-image-from-release.md

```diff
@@ -0,0 +1,105 @@
+
+For newer versions (>=4.11), on image-service startup, it would instead extract the OS image from the release. This is done by first fetching a reference to the image in the release payload, by using:
+`$ oc adm release info --image-for=machine-os-images <release-image>`
+Then, extract the image using this reference and save in the image store of the image-service:
+`$ oc image extract <image reference> --registry-config <pull-secret> --path /:<store>`
+
+As the image-service currently maintains only a list of OS images (OS_IMAGES env var), another list for release images should be introduced. So for disconnected environments, the list should contain references to the mirrored release images as done using [oc-mirror](https://docs.openshift.com/container-platform/4.10/installing/disconnected_install/installing-mirroring-disconnected.html#installing-mirroring-disconnected). 
+
+To use `oc image extract` a pull-secret is required, thus, the image-service should support handling a PULL_SECRET env var which is specified by the deployment (see 'Pull-Secret Handling' section).
+
+As for minimal ISO support, the current implementation is using an external URL to a rootfs image in the grub config. Since we now want to avoid specifying any external URLs, this flow should be modified to refer to the /boot-artifacts API instead. This API in the image-service already can serve the rootfs image directly from the OS image.
+
+### User Stories
+
+#### Story 1
+
+As an AI user deploying a disconnected on-prem environment, I want to avoid mirroring and serving OS images. Instead, rely only on the mirrored OCP release payload used for images in RELEASE_IMAGES list.
+
+### Implementation Details/Notes/Constraints
+
+#### Changes in assisted-image-service
+
+* Support a new env variable: RELEASE_IMAGES.
+  - Should be a list of OCP release images, and be in similar format to OS_IMAGES
+  - See [default_release_images.json](https://github.com/openshift/assisted-service/blob/master/data/default_release_images.json)
+* Retrieve the 'system' pull-secret and cache in memory (see 'Pull-Secret Handling' section).
+* On service startup, the following should be invoked for every release image:
+  - Fetch `machine-os-images` URI from the release.
+    E.g. 
+    `$ oc adm release info --image-for=machine-os-images quay.io/openshift-release-dev/ocp-release:4.11.0-0.nightly-multi-2022-04-18-120932-x86_64`
+    `(output) quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:0fc11beeab34f7187f6dc38ae60fb8e3e9d52d04a9a62d9f4aafe7192a7a83c1`
+  - Extract OS image from the fetched `machine-os-images`.
+    E.g. 
+    `$ oc image extract quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:0fc11beeab34f7187f6dc38ae60fb8e3e9d52d04a9a62d9f4aafe7192a7a83c1 --registry-config ~/.pullsecret --path /:/tmp/iso`
```

Comment:
You likely want ` --file=/coreos/coreos-x86_64.iso` instead of `--path` (or at least `--path /coreos:/tmp/iso`).